From 8f51d33c5d5ed58f5b7638163fa6b45bff6f9725 Mon Sep 17 00:00:00 2001
From: robo1340 <haines.todd@okstate.edu>
Date: Mon, 11 Dec 2023 18:45:37 -0600
Subject: [PATCH] added iris.dtb to Makefile

---
 arch/arm/boot/dts/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 916915e88..fb219114d 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1258,7 +1258,8 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157f-ed1.dtb \
 	stm32mp157f-ev1.dtb \
 	stm32mp157f-ev1-a7-examples.dtb \
-	stm32mp157f-ev1-m4-examples.dtb
+	stm32mp157f-ev1-m4-examples.dtb \
+	iris.dtb
 dtb-$(CONFIG_MACH_SUN4I) += \
 	sun4i-a10-a1000.dtb \
 	sun4i-a10-ba10-tvbox.dtb \
